### DISCUSSION TITLE
How to disable "Unused label" messages (neovim)?

### DISCUSSION THREAD

**Discussion by firubat** [latex-lsp/texlab#1004] [Q&A] [answered]
How do I disable "Unused label" messages (neovim)?
In general I don't know anything about configuring an LSP, and was a bit baffled by the wiki, so I'm not sure how can I disable/enable features, where can I see a list of features, etc.

**Comment by pfoerster** [comment]
@firubat You can use `texlab.diagnostics.ignoredPatterns`:

```jsonc
{
    "texlab.diagnostics.ignoredPatterns": ["Unused label"]
}
```

**Reply by firubat** [reply]
Thanks (and sorry for taking a long time), do you know where and how I insert it? I'm using LazyVim, and the config is:
```
{
  "neovim/nvim-lspconfig",
  optional = true,
  opts = {
    servers = {
      texlab = {
        keys = {
          { "<Leader>K", "<plug>(vimtex-doc-package)", desc = "Vimtex Docs", silent = true },
        },
      },
    },
  },
}
```
(I tried various ways but they didn't work...)

**Reply by clason** [reply]
That's a LazyVim issue; please ask in their support channel. The Lua table form of this snippet is
```lua
texlab = {
  diagnostics = {
    ignoredPatterns = { 'Unused label' }
  }
}
```

**Reply by firubat** [reply]
@clason  Thanks, I wasn't sure what's the right translation to lua. 
Unfortunately it doesn't work yet (but at least I don't get lsp errors on my config...), I'll ask in the LazyVim forum as well.

**Reply by firubat** [reply, marked as answer]
OK so after [asking there](https://github.com/LazyVim/LazyVim/discussions/2523), the configuration in LazyVim should be:

```
{
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        texlab = {
          settings = {
            texlab = {
              diagnostics = {
                ignoredPatterns = { "Unused label"},
              },
            },
          },
        },
      },
    },
  },
```

For me it worked only after restarting neovim.
